### PR TITLE
Fix bug that would withdraw unrelated placements

### PR DESCRIPTION
**Pull request by davidatkinsuk**
The findAllAdhocOrUnknownByApplication was not filtering out bookings not related to the application.

### CODE REVIEW

**Review by edavey — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/BookingEntity.kt [hunk 1]
```diff
@@ -133,7 +133,7 @@ interface BookingRepository : JpaRepository<BookingEntity, UUID> {
   fun findAllByCrn(crn: String): List<BookingEntity>
 
   @Query(
-    "SELECT b From BookingEntity b WHERE b.adhoc IS TRUE or b.adhoc IS NULL",
+    "SELECT b From BookingEntity b WHERE b.application = :application AND b.adhoc IS TRUE or b.adhoc IS NULL",
```
Comment (pezholio) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/BookingEntity.kt:136]: Should we have brackets here? i.e:

```suggestion
    "SELECT b From BookingEntity b WHERE b.application = :application AND (b.adhoc IS TRUE or b.adhoc IS NULL)",
```

File: src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/WithdrawalTest.kt [hunk 2]
```diff
@@ -628,6 +638,16 @@ class WithdrawalTest : IntegrationTestBase() {
               adhoc = null,
             )
 
+            // regression test to ensure other application's
```
Comment (pezholio) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/WithdrawalTest.kt:641]: Could we create a booking with a `null` adhoc too?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/BookingEntity.kt (hunk 1 above)
Comment (pezholio) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/BookingEntity.kt:136]: Or maybe `IS NOT FALSE` would cover both bases?

**Review by pezholio — CHANGES_REQUESTED** [review]
I still think this will return all bookings where `adhoc` is not present

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/BookingEntity.kt (hunk 1 above)
Comment (davidatkinsuk) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/BookingEntity.kt]: great spot, should be fixed now

**Review by pezholio — APPROVED** [review]

**Pull request merged**